pkg/oc/cli/admin/release/info.go

```diff
@@ -0,0 +1,566 @@
+	if diff.To.Digest == diff.From.Digest {
+		fmt.Fprintf(out, "Releases are identical\n")
+		return nil
+	}
+	w := tabwriter.NewWriter(out, 0, 4, 2, ' ', 0)
+	defer w.Flush()
+	now := time.Now()
+	fmt.Fprintf(w, "\tFROM\tTO\n")
+	fmt.Fprintf(w, "Name:\t%s\t%s\n", diff.From.PreferredName(), diff.To.PreferredName())
+	fmt.Fprintf(w, "Created:\t%s\t%s\n", duration.ShortHumanDuration(now.Sub(diff.From.Config.Created)), duration.ShortHumanDuration(now.Sub(diff.To.Config.Created)))
+	if from, to := diff.From.Platform(), diff.To.Platform(); from != to {
+		fmt.Fprintf(w, "OS/Arch:\t%s\t%s\n", from, to)
+	}
+
+	switch {
+	case diff.From.Metadata != nil && diff.To.Metadata != nil:
+		fmt.Fprintln(w)
+		fmt.Fprintf(w, "Version:\t%s\t%s\n", diff.From.Metadata.Version, diff.To.Metadata.Version)
+		canUpgrade := "No"
+		if stringArrContains(diff.To.Metadata.Previous, diff.From.Metadata.Version) {
+			canUpgrade = "Yes"
+		}
+		fmt.Fprintf(w, "Upgrade From:\t\t%s\n", canUpgrade)
+	case diff.From.Metadata != nil && diff.To.Metadata == nil:
+		fmt.Fprintf(w, "Has Release Metadata:\tYes\t\n")
+	case diff.From.Metadata == nil && diff.To.Metadata != nil:
+		fmt.Fprintf(w, "Has Release Metadata:\t\tYes\n")
+	}
+
+	if len(diff.ChangedImages) > 0 {
+		var keys []string
+		for k := range diff.ChangedImages {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		writeTabSection(w, func(w io.Writer) {
+			count := 0
+			for _, k := range keys {
+				if image := diff.ChangedImages[k]; image.To != nil && image.From != nil {
+					if count == 0 {
+						fmt.Fprintln(w)
+						fmt.Fprintf(w, "Images Changed:\n")
+					}
+					count++
+					old, new := digestOrRef(image.From.From.Name), digestOrRef(image.To.From.Name)
+					if old != new {
+						if showCommit {
+							fmt.Fprintf(w, "  %s\t%s\n", image.Name, gitDiffOrCommit(image.From, image.To))
+						} else {
+							fmt.Fprintf(w, "  %s\t%s\t%s\n", image.Name, old, new)
+						}
+					}
+				}
+			}
+		})
+
+		writeTabSection(w, func(w io.Writer) {
+			count := 0
+			for _, k := range keys {
+				if image := diff.ChangedImages[k]; image.From == nil {
+					if count == 0 {
+						fmt.Fprintln(w)
+						fmt.Fprintf(w, "Images Added:\n")
+					}
+					count++
+					if showCommit {
+						fmt.Fprintf(w, "  %s\t%s\n", image.Name, repoAndCommit(image.To))
+					} else {
+						fmt.Fprintf(w, "  %s\t%s\n", image.Name, digestOrRef(image.To.From.Name))
+					}
+				}
+			}
+		})
+
+		writeTabSection(w, func(w io.Writer) {
+			count := 0
+			for _, k := range keys {
+				if image := diff.ChangedImages[k]; image.To == nil {
+					if count == 0 {
+						fmt.Fprintln(w)
+						fmt.Fprintf(w, "Images Removed:\n")
+					}
+					count++
+					fmt.Fprintf(w, "  %s\n", image.Name)
+				}
+			}
+		})
+	}
+	fmt.Fprintln(w)
+	return nil
+}
+
+func repoAndCommit(ref *imageapi.TagReference) string {
+	repo := ref.Annotations["io.openshift.build.source-location"]
+	commit := ref.Annotations["io.openshift.build.commit.id"]
```

Comment:
Apparently OCP releases have this information at `io.openshift.source-commit-url`, `io.openshift.source-repo-commit`, and `io.openshift.source-repo-url`, which means `--commits` doesn't show their repository (although somehow it is showing the commit ID).  It would be super-awesome if we used [the standard Open Containers annotations][1] for this information instead of custom, per-channel annotations (we can obviously continue to set custom, per-channel annotations in parallel).

[1]: https://github.com/opencontainers/image-spec/blame/v1.0.1/annotations.md#L22-L28

Reply:
Oooooh. I like standards. @adammhaile that would be _super_ easy to slide into our images, wouldn't it?

Reply:
@tbielawa I literally just have to change 2 strings in doozer. So yes. 
I'm already doing that... right now

Reply:
> org.opencontainers.image.source and org.opencontainers.image.revision seem to be the closest match to url and commit id

Yeah.

Reply:
I've filed #21786 to support the OCI annotations for these properties, falling back to these OpenShift-specific annotations when the OCI annotations are unset.